Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -12,7 +12,7 @@ For Android, we define "root detection" a bit more broadly, including custom ROM
 
 In the following section, we list some common root detection methods you'll encounter. You'll find some of these methods implemented in the [crackme examples](https://github.com/OWASP/owasp-mstg/blob/master/OMTG-Files/02_Crackmes/List_of_Crackmes.md "OWASP Mobile Crackmes") that accompany the OWASP Mobile Testing Guide.
 
-Root detection can also be implemented through libraries such as [RootBeer](https://github.com/scottyab/rootbeer "RootBeer").
+Root detection can also be implemented through libraries such as [RootBeer](https://github.com/scottyab/rootbeer "RootBeer") or the native way ([here is an example](https://github.com/DimaKoz/meat-grinder)) .

```

Comment:
Rootbeer also has some native checks https://github.com/scottyab/rootbeer/blob/master/rootbeerlib/src/main/java/com/scottyab/rootbeer/RootBeerNative.java

Maybe you can write:
Another root detection implementation is called meat-grinder and can be found [here](https://github.com/DimaKoz/meat-grinder "meat-grinder")

Reply:
Rootbeer(it's a good library, I used it before) does native searching for some files only:

`public native int checkForRoot(Object[] pathArray)`

 but my implementation does comprehensive native checks as you can see [here:](
https://github.com/DimaKoz/meat-grinder/blob/be97a3d03ed7892a935a032334bcae0521bae0f0/app/src/main/java/com/kozhevin/rootchecks/util/MeatGrinder.java#L55-L79)

```c
    public native boolean isDetectedDevKeys();
    public native boolean isDetectedTestKeys();
    public native boolean isNotFoundReleaseKeys();
    public native boolean isFoundDangerousProps();
    public native boolean isPermissiveSelinux();
    public native boolean isSuExists();
    public native boolean isAccessedSuperuserApk();
    public native boolean isFoundSuBinary();
    public native boolean isFoundBusyboxBinary();
    public native boolean isFoundXposed();
    public native boolean isFoundResetprop();
    public native boolean isFoundWrongPathPermission();
    public native boolean isFoundHooks();

```
This project doesn't use java for getting results.
The advantage of these checks is that they successfully resist tools such as [rootcloak](https://github.com/devadvance/rootcloak#rootcloak)
Moreover, there are user reviews that my tool 

> even catches magisk

See comments [here](https://stackoverflow.com/a/46332386/3166697)

@sushi2k So what do you think about this phrase:
Another native root detection( without java) implementation is called meat-grinder and can be found [here](https://github.com/DimaKoz/meat-grinder)

Reply:
Fine for me. But you can skip the without java part, as native does anyway exclude that 😉